src/main/java/mate/academy/spring/controller/ShoppingCartController.java

```diff
@@ -0,0 +1,46 @@
+package mate.academy.spring.controller;
+
+import mate.academy.spring.model.dto.response.ShoppingCartResponseDto;
+import mate.academy.spring.service.MovieSessionService;
+import mate.academy.spring.service.ShoppingCartService;
+import mate.academy.spring.service.UserService;
+import mate.academy.spring.service.dto.mapping.impl.response.ShoppingCartResponseMapper;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/shopping-carts")
+public class ShoppingCartController {
+    private ShoppingCartService shoppingCartService;
+    private ShoppingCartResponseMapper shoppingCartResponseMapper;
+    private UserService userService;
+    private MovieSessionService movieSessionService;
+
+    public ShoppingCartController(ShoppingCartService shoppingCartService,
+                                  ShoppingCartResponseMapper shoppingCartResponseMapper,
+                                  UserService userService,
+                                  MovieSessionService movieSessionService) {
+        this.shoppingCartService = shoppingCartService;
+        this.shoppingCartResponseMapper = shoppingCartResponseMapper;
+        this.userService = userService;
+        this.movieSessionService = movieSessionService;
+    }
+
+    @GetMapping("/by-user")
+    public ShoppingCartResponseDto get(@RequestParam Long userId) {
+        return shoppingCartResponseMapper.toDto(shoppingCartService
+                .getByUser(userService.getById(userId)));
+    }
+
+    @PostMapping("/movie-sessions")
+    public ShoppingCartResponseDto addMovieSession(@RequestParam Long userId,
+                                                   @RequestParam Long movieSessionId) {
+        shoppingCartService.addSession(movieSessionService
+                .get(movieSessionId), userService.getById(userId));
+        return shoppingCartResponseMapper.toDto(shoppingCartService
+                .getByUser(userService.getById(userId)));
```

Comment:
```suggestion
        User user = userService.getById(userId);
        shoppingCartService.addSession(movieSessionService
                .get(movieSessionId), user);
        return shoppingCartResponseMapper.toDto(shoppingCartService
                .getByUser(user));
```

This way you can reduce the number of queries to the database